probe-rs/src/flashing/flasher.rs

```diff
@@ -1207,6 +1237,63 @@ impl ActiveFlasher<'_, '_, Erase> {
             Ok(())
         }
     }
+
+    pub(super) fn blank_check(&mut self, sector: &FlashSector) -> Result<(), FlashError> {
+        let address = sector.address();
+        let size = sector.size();
+        tracing::info!(
+            "Checking for blanked flash between address {:#010x} and {:#010x}",
+            address,
+            address + size
+        );
+        let t1 = Instant::now();
+
+        if let Some(blank_check) = self.flash_algorithm.pc_blank_check {
+            let error_code = self.call_function_and_wait(
+                &Registers {
+                    pc: into_reg(blank_check)?,
+                    r0: Some(into_reg(address)?),
+                    r1: Some(into_reg(size)?),
+                    r2: None,
+                    r3: None,
+                },
+                false,
+                Duration::from_millis(
+                    // self.flash_algorithm.flash_properties.erase_sector_timeout as u64,
+                    10_000,
+                ),
+            )?;
+            tracing::info!(
+                "Done checking blank. Result is {}. This took {:?}",
+                error_code,
+                t1.elapsed()
+            );
+
+            if error_code != 0 {
+                Err(FlashError::RoutineCallFailed {
+                    name: "blank_check",
+                    error_code,
+                })
+            } else {
+                self.progress.sector_erased(sector.size(), t1.elapsed());
+                Ok(())
+            }
+        } else {
+            let mut data = vec![0; size as usize];
+            self.core
+                .read(address, &mut data)
+                .map_err(FlashError::Core)?;
+            if !data
+                .iter()
+                .all(|v| *v == self.flash_algorithm.flash_properties.erased_byte_value)
```

Comment:
Just want to note that there are a few chips that have 2-byte erase values. You don't need to do anything about those here, I think we lose that information somewhere in target-gen, but I'll leave this comment here in case someone looks at this PR.